Build error with Android SDK r22 (patch included)

I get the following error when trying to compile the current master branch (d659251):

BUILD FAILED
/home/silvan/bin/adt-bundle-linux-x86_64-20130522/sdk/tools/ant/build.xml:597: The following error occurred while executing this line:
/home/silvan/neo4j/neo4j-mobile-android/neo4j-android/build.xml:753: The following error occurred while executing this line:
/home/silvan/neo4j/neo4j-mobile-android/neo4j-android/build.xml:773: renderscript doesn't support the "executable" attribute

Setup:
Ubuntu 13.04 64 Bit
Java(TM) SE Runtime Environment (build 1.7.0_21-b11)
Android SDK r22.0.5
Android SDK Build-tools 17
Apache Ant(TM) version 1.8.2 compiled on May 18 2012

I was able to resolve all compile errors by modifying the build.xml file in the neo4j-android folder as follows:

diff --git a/neo4j-android/build.xml b/neo4j-android/build.xml
index 5196969..b501d5b 100644
--- a/neo4j-android/build.xml
+++ b/neo4j-android/build.xml
@@ -280,9 +280,6 @@
     <condition property="bat" value=".bat" else=""><os family="windows" /></condition>
     <property name="adb" location="${android.platform.tools.dir}/adb${exe}" />
     <property name="zipalign" location="${android.tools.dir}/zipalign${exe}" />
-    <property name="aidl" location="${android.platform.tools.dir}/aidl${exe}" />
-    <property name="aapt" location="${android.platform.tools.dir}/aapt${exe}" />
-    <property name="dx" location="${android.platform.tools.dir}/dx${bat}" />
   <property name="renderscript" location="${android.platform.tools.dir}/llvm-rs-cc${exe}"/>
   <property name="lint" location="${android.tools.dir}/lint${bat}" />

@@ -590,6 +587,11 @@

```
 <!-- Pre build setup -->
 <target name="-build-setup" depends="-setup">
```
-        <!-- find location of build tools -->
-        <getbuildtools name="android.build.tools.dir" />
-        <property name="aidl" location="${android.build.tools.dir}/aidl${exe}" />
-        <property name="aapt" location="${android.build.tools.dir}/aapt${exe}" />
- ```
     <property name="dx" location="${android.build.tools.dir}/dx${bat}" />
  
   <!-- read the previous build mode -->
   <property file="${out.build.prop.file}" />
  ```
  
  @@ -763,7 +765,7 @@
           <!-- renderscript generates resources so it must be called before aapt -->
           <echo level="info">----------</echo>
           <echo level="info">Handling RenderScript files...</echo>
-            <renderscript executable="${renderscript}"
-            <renderscript buildToolsRoot="${android.build.tools.dir}"
                   includePathRefId="android.renderscript.include.path"
                   genFolder="${gen.absolute.dir}"
                   resFolder="${out.res.absolute.dir}/raw"

--- a/neo4j-android/build.xml
+++ b/neo4j-android/build.xml
@@ -280,9 +280,6 @@
     <condition property="bat" value=".bat" else=""><os family="windows" /></condition>
     <property name="adb" location="${android.platform.tools.dir}/adb${exe}" />
     <property name="zipalign" location="${android.tools.dir}/zipalign${exe}" />
-    <property name="aidl" location="${android.platform.tools.dir}/aidl${exe}" />
-    <property name="aapt" location="${android.platform.tools.dir}/aapt${exe}" />
-    <property name="dx" location="${android.platform.tools.dir}/dx${bat}" />
     <property name="renderscript" location="${android.platform.tools.dir}/llvm-rs-cc${exe}"/>
     <property name="lint" location="${android.tools.dir}/lint${bat}" />
 
@@ -590,6 +587,11 @@
 
     <!-- Pre build setup -->
     <target name="-build-setup" depends="-setup">
+        <!-- find location of build tools -->
+        <getbuildtools name="android.build.tools.dir" />
+        <property name="aidl" location="${android.build.tools.dir}/aidl${exe}" />
+        <property name="aapt" location="${android.build.tools.dir}/aapt${exe}" />
+        <property name="dx" location="${android.build.tools.dir}/dx${bat}" />
 
         <!-- read the previous build mode -->
         <property file="${out.build.prop.file}" />
@@ -763,7 +765,7 @@
             <!-- renderscript generates resources so it must be called before aapt -->
             <echo level="info">----------</echo>
             <echo level="info">Handling RenderScript files...</echo>
-            <renderscript executable="${renderscript}"
+            <renderscript buildToolsRoot="${android.build.tools.dir}"
                     includePathRefId="android.renderscript.include.path"
                     genFolder="${gen.absolute.dir}"
                     resFolder="${out.res.absolute.dir}/raw"

